x-pack/plugins/event_log/README.md

```diff
@@ -1,186 +1,248 @@
-    "@timestamp": "2019-12-09T21:16:43.425Z",
-    "ecs": {
-      "version": "1.3.1"
-    }
-  }
-}
+```js
+{ ids: ["saved-object-id-1", ...] }
 ```
 
-The shape of the document written to the index is a subset of [ECS][] with an
-extended field of `kibana` with some Kibana-related properties contained within
-it.
+The response is `QueryEventsBySavedObjectResult` described below.
 
-The ES mappings for the ECS data, and the config-schema for the ECS data, are
-generated by a script, and available here:
+## Plugin Client APIs for querying
 
-- [`generated/mappings.json`](generated/mappings.json)
-- [`generated/schemas.ts`](generated/schemas.ts)
+```ts
+interface EventLogClient {
+  findEventsBySavedObjectIds(
+    type: string,
+    ids: string[],
+    options?: Partial<FindOptionsType>
+  ): Promise<QueryEventsBySavedObjectResult>;
+}
 
-It's anticipated that these interfaces will grow over time, hopefully adding
-more ECS fields but adding Kibana extensions as required.
+interface FindOptionsType { /* typed version of HTTP query parameters ^^^ */ }
 
-Since there are some security concerns with the data, we are currently
-restricting access via known saved object ids.  That is, you can only query
-history records associated with specific saved object ids.
+interface QueryEventsBySavedObjectResult {
+  page: number;
+  per_page: number;
+  total: number;
+  data: Event[];
+}
+```
 
-[ECS]: https://www.elastic.co/guide/en/ecs/current/index.html
+## Generating Events
 
+Follow these steps to use `eventLog` in your plugin: 
 
-## API
+1. Declare `eventLog` as a dependency in `kibana.json`:
 
-```typescript
-// IEvent is a TS type generated from the subset of ECS supported
+```json
+{
+  ...
+  "requiredPlugins": ["eventLog"],
+  ...
+}
+```
 
-// the NP plugin returns a service instance from setup() and start()
-export interface IEventLogService {
-  registerProviderActions(provider: string, actions: string[]): void;
-  isProviderActionRegistered(provider: string, action: string): boolean;
-  getProviderActions(): Map<string, Set<string>>;
+2. Register provider / actions, and create your plugin's logger, using service
+API provided in the `setup` stage:
 
-  getLogger(properties: IEvent): IEventLogger;
+```typescript
+...
+import { IEventLogger, IEventLogService } from '../../event_log/server';
+interface PluginSetupDependencies {
+  eventLog: IEventLogService;
 }
-
-export interface IEventLogger {
-  logEvent(properties: IEvent): void;
-  startTiming(event: IEvent): void;
-  stopTiming(event: IEvent): void;
+...
+public setup(core: CoreSetup, { eventLog }: PluginSetupDependencies) {
+  ...
+  eventLog.registerProviderActions('my-plugin', ['action-1, action-2']);
+  const eventLogger: IEventLogger = eventLog.getLogger({ event: { provider: 'my-plugin' } });
+  ...
 }
+...
+```
+
+4. To log an event, call `logEvent()` on the `eventLogger` object you created:
```

Comment:
Where's step 3?

Reply:
THERE IS NO STEP 3!!! :-).  

Fixed to change step 4 to step 3, my bad